lib/xgu/xgu.h

```diff
@@ -0,0 +1,937 @@
+    XGU_TEXTURE_FORMAT_COLOR_SZ_I8_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_I8_A8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_L_DXT1_A1R5G5B5 = NV097_SET_TEXTURE_FORMAT_COLOR_L_DXT1_A1R5G5B5,
+    XGU_TEXTURE_FORMAT_COLOR_L_DXT23_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_L_DXT23_A8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_L_DXT45_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_L_DXT45_A8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_A1R5G5B5 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A1R5G5B5,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_R5G6B5 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_R5G6B5,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_Y8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_Y8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_A8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_A8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_A8Y8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_A8Y8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_AY8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_AY8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_X1R5G5B5 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_X1R5G5B5,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_A4R4G4B4 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A4R4G4B4,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_X8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_X8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8Y8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8Y8,
+    XGU_TEXTURE_FORMAT_COLOR_LC_IMAGE_CR8YB8CB8YA8 = NV097_SET_TEXTURE_FORMAT_COLOR_LC_IMAGE_CR8YB8CB8YA8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_R6G5B5 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_R6G5B5,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_R8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_R8B8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_DEPTH_X8_Y24_FIXED = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_DEPTH_X8_Y24_FIXED,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_DEPTH_Y16_FIXED = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_DEPTH_Y16_FIXED,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_Y16 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_Y16,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_A8B8G8R8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_A8B8G8R8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_R8G8B8A8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_R8G8B8A8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8B8G8R8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_A8B8G8R8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_B8G8R8A8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_B8G8R8A8,
+    XGU_TEXTURE_FORMAT_COLOR_LU_IMAGE_R8G8B8A8 = NV097_SET_TEXTURE_FORMAT_COLOR_LU_IMAGE_R8G8B8A8
+} XguTexFormatColor;
+
+typedef enum {
+    XGU_PALETTE_LENGTH_256 = NV097_SET_TEXTURE_PALETTE_LENGTH_256,
+    XGU_PALETTE_LENGTH_128 = NV097_SET_TEXTURE_PALETTE_LENGTH_128,
+    XGU_PALETTE_LENGTH_64 = NV097_SET_TEXTURE_PALETTE_LENGTH_64,
+    XGU_PALETTE_LENGTH_32 = NV097_SET_TEXTURE_PALETTE_LENGTH_32,    
+} XguPaletteLen;
```

Comment:
I'm not sure how exactly this is used, but I think it might be smarter to remove this enum and just have an `unsigned int size` - then have a switch case in the function to auto-pick the correct value.

However, I'm still undecided.